src/Core/src/Core/IFrameworkElement.cs

```diff
@@ -80,13 +82,18 @@ public interface IFrameworkElement
 		void InvalidateArrange();
 
 		/// <summary>
-		/// Id used by automation tools to interact with this FrameworkElement
+		/// Id used by automation tools to interact with this FrameworkElement.
 		/// </summary>
 		string AutomationId { get; }
 
 		/// <summary>
-		/// Direction in which the UI elements on the page are scanned by the eye
+		/// Direction in which the UI elements on the page are scanned by the eye.
 		/// </summary>
 		FlowDirection FlowDirection { get; }
+
+		/// <summary>
+		/// Gets the geometry used to define the outline of the contents of a View.
+		/// </summary>
+		IGeometry Clip { get; }
```

Comment:
In Xamarin.Forms we use Geometry to allow clipping. Should we keep using Geometry (RectangleGeometry, EllipseGeometry, etc) or directly use Shapes (Rectangle, Ellipse etc)?

Reply:
Use `IShape` and Microsoft.Maui.Graphics.